app/controllers/cart_products_controller.rb

```diff
@@ -0,0 +1,48 @@
+class CartProductsController < ApplicationController
+  before_action :getProduct, only: [:new]
+  before_action :getCart, only: [:new,:destroy,:add_quantity]
+
+  def new
+    if @cart.products.include?(@product)
+      @cart_product = @cart.cart_products.find_by(:product_id => @product.id)
+      @cart_product.quantity +=1
+    else
+      @cart_product = @cart.cart_products.new
+      @cart_product.product_id = @product.id
+    end
+    @cart_product.subtotal = @product.price * @cart_product.quantity
+    @cart_product.save
+    redirect_to product_path(@product), notice: "Product add to the cart Successfully"
```

Comment:
why we're not using respond_to here?

Reply:
Why do we have to use respond_to instead of redirect_to?